mir_eval/melody.py

```diff
@@ -563,7 +563,8 @@ def raw_chroma_accuracy(ref_voicing, ref_cent, est_voicing, est_cent,
     # Raw chroma = same as raw pitch except that octave errors are ignored.
     cent_diff = np.abs(ref_cent - est_cent)
     octave = 1200*np.floor(cent_diff/1200.0 + 0.5)
-    frame_correct = (np.abs(cent_diff - octave)[ref_voicing] < cent_tolerance)
+    correct_voicing = ref_voicing * est_voicing
+    frame_correct = (np.abs(cent_diff - octave)[correct_voicing] < cent_tolerance)
```

Comment:
(cc @juanjobosch who stared at this with me!)

This almost fixes the problem, but with one annoying subtlety - f0 values in the input can be given as "negative" frequency values - in this case, the difference should be computed over all the reference frames, using the absolute value of the frequency given. It's not very explicit in the definition, but it's meant to be a way to measure pitch/chroma accuracy independent of voicing mistakes:
> Raw chroma accuracy: the fraction of voiced frames in ref_cent for which est_cent provides a correct frequency values (within cent_tolerance cents), ignoring octave errors

Consider the case:
```
ref_t = [1,2,3]
est_t = [1,2,3]
ref_f = [440, 440, 440]
est_f = [-220,-220,-220]
```
The Raw Chroma Accuracy should be 100%, even though the estimate voicing is `[0,0,0]`. 

So instead, I propose the following fix that solves the bug but doesn't break the negative frequency use case.
```
matching_voicing = ref_voicing * (est_cent > 0)
frame_correct = (np.abs(cent_diff - octave)[matching_voicing] < cent_tolerance)
```
In short, this only count frames as "correct" if the estimate is non-zero (where zero indicates an unvoiced frame where no "best guess" frequency is given) AND within the tolerance.

In fact, this bug exists for raw_pitch_accuracy too... but the "problem" case never comes up because we almost never report frequencies below 40 Hz, so these cases always fall outside of the tolerance threshold. For good measure though, we should make the same fix for raw_pitch_accuracy. (cc @craffel )